src/app/application.cpp

```diff
@@ -511,36 +511,32 @@ void Application::initializeTranslation()
 {
     Preferences* const pref = Preferences::instance();
     // Load translation
-    QString locale = pref->getLocale();
-
-    if (locale.isEmpty()) {
-        locale = QLocale::system().name();
-        pref->setLocale(locale);
-    }
+    QString localeStr = pref->getLocale();
+    QLocale::setDefault(QLocale(localeStr));
 
     if (m_qtTranslator.load(
 #ifdef QBT_USES_QT5
-            QString::fromUtf8("qtbase_") + locale, QLibraryInfo::location(QLibraryInfo::TranslationsPath)) ||
+            QString::fromUtf8("qtbase_") + localeStr, QLibraryInfo::location(QLibraryInfo::TranslationsPath)) ||
         m_qtTranslator.load(
 #endif
-            QString::fromUtf8("qt_") + locale, QLibraryInfo::location(QLibraryInfo::TranslationsPath))) {
-            qDebug("Qt %s locale recognized, using translation.", qPrintable(locale));
+            QString::fromUtf8("qt_") + localeStr, QLibraryInfo::location(QLibraryInfo::TranslationsPath))) {
+            qDebug("Qt %s localeStr recognized, using translation.", qPrintable(localeStr));
```

Comment:
Is it really needed to replace "locale" with "localeStr" in the debug message hereinafter? This breaks sentences.